dags/openshift_nightlies/scripts/install/rosa.sh

```diff
@@ -208,24 +210,37 @@ setup(){
         rosa whoami
         rosa verify quota
         rosa verify permissions
-        export ROSA_VERSION=$(rosa list versions -o json --channel-group=nightly | jq -r '.[] | select(.raw_id|startswith('\"${version}\"')) | .raw_id' | sort -rV | head -1)
-        [ -z "${ROSA_VERSION}" ] && echo "ERROR: Image not found for version (${version}) on ROSA Nightly channel group" && exit 1
-        return 0        
+        if [ "${ROSA_OCP_VERSION}" == "latest" ] ; then
+            export ROSA_VERSION=$(rosa list versions -o json --channel-group=${ROSA_CHANNEL_GROUP} | jq -r '.[] | select(.raw_id|startswith('\"${version}\"')) | .raw_id' | sort -rV | head -1)
+        else
+            export ROSA_VERSION=$(rosa list versions -o json --channel-group=${ROSA_CHANNEL_GROUP} | jq -r '.[] | select(.raw_id|startswith('\"${version}\"')) | .raw_id' | grep ^${ROSA_OCP_VERSION}$)
+        fi
+        [ -z "${ROSA_VERSION}" ] && echo "ERROR: Image not found for version (${version}) on ROSA ${ROSA_CHANNEL_GROUP} channel group" && exit 1
+        if [ "${ROSA_CHANNEL_GROUP}" == "nightly" ] ; then
+            ROSA_VERSION="${ROSA_VERSION}-nightly"
+        elif [ "${ROSA_CHANNEL_GROUP}" == "candidate" ] ; then
+            ROSA_VERSION="${ROSA_VERSION}-candidate"
+        fi
+        return 0
     fi
 }
 
 install(){
     export COMPUTE_WORKERS_TYPE=$(cat ${json_file} | jq -r .openshift_worker_instance_type)
     if [[ $INSTALL_METHOD == "osd" ]]; then
-        export OCM_VERSION=$(ocm list versions --channel-group nightly | grep ^${version} | sort -rV | head -1)
-        [ -z ${OCM_VERSION} ] && echo "ERROR: Image not found for version (${version}) on OCM Nightly channel group" && exit 1
-        ocm create cluster --ccs --provider aws --region ${AWS_REGION} --aws-account-id ${AWS_ACCOUNT_ID} --aws-access-key-id ${AWS_ACCESS_KEY_ID} --aws-secret-access-key ${AWS_SECRET_ACCESS_KEY} --channel-group nightly --version ${OCM_VERSION} --multi-az --compute-nodes ${COMPUTE_WORKERS_NUMBER} --compute-machine-type ${COMPUTE_WORKERS_TYPE} --network-type ${NETWORK_TYPE} ${CLUSTER_NAME}
+	if [ "${ROSA_OCP_VERSION}" == "latest" ] ; then
```

Comment:
Why are we using a ROSA_OCP_VERSION here and not OCM_VERSION?

Reply:
Parameter on config files is rosa_ocp_version, it haven't been duplicated for OCM, as it points to an OCP version.
If pointed to a fixed version, script will check if version exists using ocm list version or rosa list version depending on the $cluster_install_method var, if points to latest, it will get the latest image available on $rosa_channel_group channel.

Reply:
If we are using the variable generically then we should change the name of it from ROSA.... to something else. MANAGED_OCP_VERSION perhaps?

Reply:
modified to MANAGED_OCP_VERSION. in the same way I have changed ROSA_CHANNEL_GROUP to MANAGED_CHANNEL_GROUP